### PR TITLE
Fix Fuse open read_write then truncate to extend file length

### LINKED ISSUE #15852: Fuse create -> open(read_write) -> truncate()
**Alluxio Version:**
Master

**Describe the bug**
pjdfstest ftruncate/00.t
```
expect 0 create ${n0} 0644
expect 0 open ${n0} O_RDWR : ftruncate 0 1234567
expect 1234567 lstat ${n0} size
expected 1234567, got 0
```
```
Enter: Fuse.Open(path=/pjdfstest_2755a4e04206b24cccf519a611bd5596/pjdfstest_1ad7b004f3b6a00e82ab9c3aecf2a7db,flags=0x8002)
2022-07-07 23:20:04,434 DEBUG AlluxioJniFuseFileSystem - Exit (0): Fuse.Truncate(path=/pjdfstest_2755a4e04206b24cccf519a611bd5596/pjdfstest_1ad7b004f3b6a00e82ab9c3aecf2a7db,size=1234567) in 0 ms
```

**To Reproduce**
Steps to reproduce the behavior (as minimally and precisely as possible)

**Expected behavior**

**Urgency**
Describe the impact and urgency of the bug.

**Are you planning to fix it**
Please indicate if you are already working on a PR. 

**Additional context**

### PR DESCRIPTION
**Pull request by LuQQiu**
### What changes are proposed in this pull request?

Fixes #15852 support create empty file -> open(rdwr) -> truncate() workload

### CODE REVIEW

File: integration/fuse/src/main/java/alluxio/fuse/file/FuseFileInOrOutStream.java [hunk 1]
```diff
@@ -158,14 +158,21 @@ public synchronized void truncate(long size) {
...
       return;
     }
-    if (size == getFileLength()) {
+    long currentSize = getFileLength();
+    if (size == currentSize) {
       return;
     }
-    if (size == 0) {
+    if (size == 0 || currentSize == 0) {
       AlluxioFuseUtils.deletePath(mFileSystem, mUri);
```
Comment (yyongycy) [inline integration/fuse/src/main/java/alluxio/fuse/file/FuseFileInOrOutStream.java:166]: are we sure there is no current fd opened for this path?

**Comment by LuQQiu** [comment]
@yyongycy @ssz1997 PTAL, thanks

File: integration/fuse/src/main/java/alluxio/fuse/file/FuseFileInOrOutStream.java (hunk 1 above)
Comment (yyongycy) [reply integration/fuse/src/main/java/alluxio/fuse/file/FuseFileInOrOutStream.java:166]: This is an extreme case, like two threads:
one is opening one file, with fd1
then second one is opening it again, with fd2.

This deletion will directly delete the file, that will leave the fd1 dangling.

I am wondering if this scenario exists. This change looks good to me.

Comment (ssz1997) [reply integration/fuse/src/main/java/alluxio/fuse/file/FuseFileInOrOutStream.java:166]: I think this create() -> open() -> truncate() is basically creating a file. Your scenario is valid, but I don't think in real life there's workload accessing the file while creating it.

**Review by ssz1997 — APPROVED** [review]
LGTM

File: integration/fuse/src/main/java/alluxio/fuse/file/FuseFileInOrOutStream.java (hunk 1 above)
Comment (ssz1997) [reply integration/fuse/src/main/java/alluxio/fuse/file/FuseFileInOrOutStream.java:166]: From my understanding truncate is almost only used for testing purpose?

**Review by yyongycy — APPROVED** [review]
LGTM

**Comment by yyongycy** [comment]
alluxio-bot, merge this please

**Pull request merged**